### ISSUE TITLE
RuboCop wrongly disallows building times with `find_zone` constructs in Rails code

### ISSUE THREAD

**Issue by attilahorvath** [bbatsov/rubocop#4047]
In my codebase, I'm using the following constructs to build times in specified timezones:

```ruby
Time.find_zone('EST').local(2017, 2, 17, 8, 0)

# and

Time.find_zone('EST').parse('10am')
```

However, RuboCop forbids me from using these, wrongly assuming that time zone information would be lost this way.

This is not the case, the resulting objects are of type `ActiveSupport::TimeWithZone` with the correct time zone set on them.

--------

## Expected behavior

RuboCop shouldn't warn against using constructs with `find_zone`.

## Actual behavior

I am getting RuboCop warnings. As an example:

```
spec/models/my_model_spec.rb:39:24: C: Do not use Time.local without zone. Use one of Time.zone.local, Time.current, Time.local.in_time_zone, Time.local.utc, Time.local.getlocal, Time.local.iso8601, Time.local.jisx0301, Time.local.rfc3339, Time.local.to_i, Time.local.to_f instead.
        travel_to Time.find_zone('EST').local(2017, 2, 17, 8, 0) do
```

## Steps to reproduce the problem

Write a code file containing similar constructs to the ones I've showed and run RuboCop with default settings against the file.

## RuboCop version

```
$ bundle exec rubocop -V
0.47.1 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-linux)
```

**Comment by attilahorvath** [comment]
Submitted a pull request to fix this issue: https://github.com/bbatsov/rubocop/pull/4054
